background.js

```diff
@@ -1 +1,58 @@
-//TO DO?
\ No newline at end of file
+const Color = {
+  light: 0,
+  dark: 1
+};
+const threshold = 220;
+
+// The simplest method for now. 
+// It takes the colors from all fill, stroke or style attributes, 
+// finds the middle value and calculates accordingly.
+function weightedColor(svgElement) {
+  var colors = svgElement.match(/(style|fill|stroke)="#[0-9a-fA-F]{3,6}|rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)*"/g);
+
+  if (colors == null) {
+    return Color.dark;
+  }
+
+  var midValue = 0;
+  for (const color of colors) {
+    // TODO: Add a weight based on the area of current path.
```

Comment:
İçinde eğriler olan poligonlar için burada değişik şeyler yapmak gerekebilir. Şimdilik TODO bırakmak mantıklı.